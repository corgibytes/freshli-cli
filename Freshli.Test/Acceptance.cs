using System;
using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalTests.Reporters.TestFrameworks;
using Freshli.Languages.Php;
using Freshli.Languages.Ruby;
using Freshli.Languages.Python;
using Xunit;

namespace Freshli.Test
{
  [UseReporter(typeof(XUnit2Reporter))]
  public class Acceptance
  {
    public Acceptance() {
      ManifestFinder.Register<RubyBundlerManifestFinder>();
      ManifestFinder.Register<PhpComposerManifestFinder>();
      ManifestFinder.Register<PipRequirementsTxtManifestFinder>();

      FileHistoryFinder.Register<GitFileHistoryFinder>();
    }

    private DateTime _testingBoundary =
      new DateTime(2020, 01, 01, 0, 0, 0, DateTimeKind.Utc);

    [Fact]
    public void RubyGemsWithGitHistory()
    {
      var runner = new Runner();

      var rubyFixturePath = Fixtures.Path("ruby", "nokotest");
      var results = runner.Run(rubyFixturePath, asOf: _testingBoundary);

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void RubyGemsWithHistoryViaGitHub() {
      var runner = new Runner();

      var repoUrl =
        "https://github.com/corgibytes/freshli-fixture-ruby-nokotest";
      var results = runner.Run(repoUrl, asOf: _testingBoundary);

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void RubyGemsFeedbinHistoryViaGitHub()
    {
      var runner = new Runner();

      var repoUrl = "https://github.com/feedbin/feedbin";
      var results = runner.Run(repoUrl, asOf: _testingBoundary);

      Approvals.VerifyAll(results, "results");
    }

    [Fact] 
    public void RubyGemsClearanceHistoryViaGitHub()
    {
      ManifestFinder.Register<RubyBundlerManifestFinder>();
      FileHistoryFinder.Register<GitFileHistoryFinder>();

      var runner = new Runner();
      var results = runner.Run("https://github.com/thoughtbot/clearance");
      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void ComposerWithoutGitHistory()
    {
      var runner = new Runner();

      var phpFixturePath = Fixtures.Path("php", "large");
      var results = runner.Run(phpFixturePath, asOf: _testingBoundary);

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void DrupalComposerWithoutGitHistory()
    {
      var runner = new Runner();

      var phpFixturePath = Fixtures.Path("php", "drupal");
      var results = runner.Run(phpFixturePath, asOf: _testingBoundary);

      Approvals.VerifyAll(results, "results");
    }

    [Fact]
    public void RequirementsTxtPyspider() {
      var runner = new Runner();

      var results = runner.Run(
        "https://github.com/binux/pyspider",
        asOf: _testingBoundary
      );

      Approvals.VerifyAll(results, "results");
    }
  }
}
