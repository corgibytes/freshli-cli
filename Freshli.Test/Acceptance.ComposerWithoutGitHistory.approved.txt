results[0] = { Date: 2020-01-01T00:00:00, LibYear: { _packagesValues: [ { Name: "aws/aws-sdk-php", Version: "3.130.2", PublishedAt: 2019-12-23T00:00:00, Value: 0 }, { Name: "barryvdh/laravel-dompdf", Version: "v0.8.5", PublishedAt: 2019-08-23T00:00:00, Value: 0 }, { Name: "barryvdh/laravel-snappy", Version: "v0.4.6", PublishedAt: 2019-10-02T00:00:00, Value: 0 }, { Name: "cogpowered/finediff", Version: "0.3.1", PublishedAt: 2014-05-19T00:00:00, Value: 0 }, { Name: "doctrine/cache", Version: "1.10.0", PublishedAt: 2019-11-29T00:00:00, Value: 0 }, { Name: "doctrine/dbal", Version: "v2.10.0", PublishedAt: 2019-11-03T00:00:00, Value: 0 }, { Name: "doctrine/event-manager", Version: "1.1.0", PublishedAt: 2019-11-10T00:00:00, Value: 0 }, { Name: "doctrine/inflector", Version: "1.3.1", PublishedAt: 2019-10-30T00:00:00, Value: 0 }, { Name: "doctrine/lexer", Version: "1.2.0", PublishedAt: 2019-10-30T00:00:00, Value: 0 }, { Name: "dompdf/dompdf", Version: "v0.8.3", PublishedAt: 2018-12-14T00:00:00, Value: 0 }, { Name: "dragonmantank/cron-expression", Version: "v2.3.0", PublishedAt: 2019-03-31T00:00:00, Value: 0 }, { Name: "egulias/email-validator", Version: "2.1.13", PublishedAt: 2019-12-30T00:00:00, Value: 0.0273972602739726 }, { Name: "erusev/parsedown", Version: "1.7.4", PublishedAt: 2019-12-30T00:00:00, Value: 0.7890410958904109 }, { Name: "fideloper/proxy", Version: "4.2.2", PublishedAt: 2019-12-20T00:00:00, Value: 0 }, { Name: "gathercontent/htmldiff", Version: "0.2.1", PublishedAt: 2015-04-15T00:00:00, Value: 0 }, { Name: "guzzlehttp/guzzle", Version: "6.5.2", PublishedAt: 2019-12-23T00:00:00, Value: 0 }, { Name: "guzzlehttp/promises", Version: "v1.3.1", PublishedAt: 2016-12-20T00:00:00, Value: 0 }, { Name: "guzzlehttp/psr7", Version: "1.6.1", PublishedAt: 2019-07-01T00:00:00, Value: 0 }, { Name: "intervention/image", Version: "2.5.1", PublishedAt: 2019-11-02T00:00:00, Value: 0 }, { Name: "knplabs/knp-snappy", Version: "v1.2.0", PublishedAt: 2019-12-02T00:00:00, Value: 0 }, { Name: "laravel/framework", Version: "v6.9.0", PublishedAt: 2019-12-19T00:00:00, Value: 0 }, { Name: "laravel/socialite", Version: "v4.3.1", PublishedAt: 2019-11-26T00:00:00, Value: 0 }, { Name: "league/flysystem", Version: "1.0.62", PublishedAt: 2019-12-29T00:00:00, Value: 0.057534246575342465 }, { Name: "league/flysystem-aws-s3-v3", Version: "1.0.23", PublishedAt: 2019-06-05T00:00:00, Value: 0 }, { Name: "league/oauth1-client", Version: "1.7.0", PublishedAt: 2016-08-17T00:00:00, Value: 0 }, { Name: "monolog/monolog", Version: "2.0.2", PublishedAt: 2019-12-20T00:00:00, Value: 0 }, { Name: "mtdowling/jmespath.php", Version: "2.5.0", PublishedAt: 2019-12-30T00:00:00, Value: 3.073972602739726 }, { Name: "nesbot/carbon", Version: "2.28.0", PublishedAt: 2019-12-16T00:00:00, Value: 0 }, { Name: "onelogin/php-saml", Version: "3.4.1", PublishedAt: 2019-11-25T00:00:00, Value: 0 }, { Name: "opis/closure", Version: "3.5.1", PublishedAt: 2019-11-29T00:00:00, Value: 0 }, { Name: "paragonie/random_compat", Version: "v9.99.99", PublishedAt: 2018-07-02T00:00:00, Value: 0 }, { Name: "phenx/php-font-lib", Version: "0.5.1", PublishedAt: 2017-09-13T00:00:00, Value: 0 }, { Name: "phenx/php-svg-lib", Version: "v0.3.3", PublishedAt: 2019-09-11T00:00:00, Value: 0 }, { Name: "phpoption/phpoption", Version: "1.7.2", PublishedAt: 2019-12-15T00:00:00, Value: 0 }, { Name: "predis/predis", Version: "v1.1.1", PublishedAt: 2016-06-16T00:00:00, Value: 0 }, { Name: "psr/container", Version: "1.0.0", PublishedAt: 2017-02-14T00:00:00, Value: 0 }, { Name: "psr/http-message", Version: "1.0.1", PublishedAt: 2016-08-06T00:00:00, Value: 0 }, { Name: "psr/log", Version: "1.1.2", PublishedAt: 2019-11-01T00:00:00, Value: 0 }, { Name: "psr/simple-cache", Version: "1.0.1", PublishedAt: 2017-10-23T00:00:00, Value: 0 }, { Name: "ralouphie/getallheaders", Version: "3.0.3", PublishedAt: 2019-03-08T00:00:00, Value: 0 }, { Name: "ramsey/uuid", Version: "3.9.2", PublishedAt: 2019-12-17T00:00:00, Value: 0 }, { Name: "robrichards/xmlseclibs", Version: "3.0.4", PublishedAt: 2019-11-05T00:00:00, Value: 0 }, { Name: "sabberworm/php-css-parser", Version: "8.3.0", PublishedAt: 2019-02-22T00:00:00, Value: 0 }, { Name: "socialiteproviders/discord", Version: "v2.0.2", PublishedAt: 2018-05-26T00:00:00, Value: 0 }, { Name: "socialiteproviders/gitlab", Version: "v3.1", PublishedAt: 2018-06-27T00:00:00, Value: 0 }, { Name: "socialiteproviders/manager", Version: "v3.4.3", PublishedAt: 2019-09-25T00:00:00, Value: 0 }, { Name: "socialiteproviders/microsoft-azure", Version: "v3.0.0", PublishedAt: 2017-01-25T00:00:00, Value: 0 }, { Name: "socialiteproviders/okta", Version: "v1.1.0", PublishedAt: 2019-09-06T00:00:00, Value: 1.7917808219178082 }, { Name: "socialiteproviders/slack", Version: "v3.1", PublishedAt: 2019-01-11T00:00:00, Value: 0 }, { Name: "socialiteproviders/twitch", Version: "v5.1", PublishedAt: 2019-07-01T00:00:00, Value: 0 }, { Name: "swiftmailer/swiftmailer", Version: "v6.2.3", PublishedAt: 2019-11-12T00:00:00, Value: 0 }, { Name: "symfony/console", Version: "v5.0.2", PublishedAt: 2019-12-17T00:00:00, Value: 0 }, { Name: "symfony/css-selector", Version: "v5.0.2", PublishedAt: 2019-11-18T00:00:00, Value: 0.10136986301369863 }, { Name: "symfony/debug", Version: "v4.4.2", PublishedAt: 2019-12-16T00:00:00, Value: 0 }, { Name: "symfony/error-handler", Version: "v5.0.2", PublishedAt: 2019-12-16T00:00:00, Value: 0 }, { Name: "symfony/event-dispatcher", Version: "v5.0.2", PublishedAt: 2019-11-18T00:00:00, Value: -0.0273972602739726 }, { Name: "symfony/event-dispatcher-contracts", Version: "v2.0.1", PublishedAt: 2019-11-18T00:00:00, Value: 0.16986301369863013 }, { Name: "symfony/finder", Version: "v5.0.2", PublishedAt: 2019-11-18T00:00:00, Value: 0.0027397260273972603 }, { Name: "symfony/http-foundation", Version: "v5.0.2", PublishedAt: 2019-12-19T00:00:00, Value: 0 }, { Name: "symfony/http-kernel", Version: "v5.0.2", PublishedAt: 2019-12-19T00:00:00, Value: 0 }, { Name: "symfony/mime", Version: "v5.0.2", PublishedAt: 2019-11-30T00:00:00, Value: 0 }, { Name: "symfony/polyfill-ctype", Version: "v1.13.2", PublishedAt: 2019-11-27T00:00:00, Value: 0 }, { Name: "symfony/polyfill-iconv", Version: "v1.13.2", PublishedAt: 2019-12-03T00:00:00, Value: 0.01643835616438356 }, { Name: "symfony/polyfill-intl-idn", Version: "v1.13.2", PublishedAt: 2019-11-27T00:00:00, Value: 0 }, { Name: "symfony/polyfill-mbstring", Version: "v1.13.1", PublishedAt: 2019-11-27T00:00:00, Value: 0 }, { Name: "symfony/polyfill-php72", Version: "v1.13.2", PublishedAt: 2019-11-27T00:00:00, Value: 0 }, { Name: "symfony/polyfill-php73", Version: "v1.13.2", PublishedAt: 2019-11-27T00:00:00, Value: 0 }, { Name: "symfony/process", Version: "v5.0.2", PublishedAt: 2019-12-10T00:00:00, Value: 0.010958904109589041 }, { Name: "symfony/routing", Version: "v5.0.2", PublishedAt: 2019-12-12T00:00:00, Value: 0 }, { Name: "symfony/service-contracts", Version: "v2.0.1", PublishedAt: 2019-11-18T00:00:00, Value: 0.0958904109589041 }, { Name: "symfony/translation", Version: "v5.0.2", PublishedAt: 2019-12-12T00:00:00, Value: 0 }, { Name: "symfony/translation-contracts", Version: "v2.0.1", PublishedAt: 2019-11-18T00:00:00, Value: 0.16986301369863013 }, { Name: "symfony/var-dumper", Version: "v5.0.2", PublishedAt: 2019-12-18T00:00:00, Value: 0 }, { Name: "tijsverkoyen/css-to-inline-styles", Version: "2.2.2", PublishedAt: 2019-10-24T00:00:00, Value: 0 }, { Name: "vlucas/phpdotenv", Version: "v4.1.0", PublishedAt: 2019-12-14T00:00:00, Value: 0.2602739726027397 }, { Name: "barryvdh/laravel-debugbar", Version: "v3.2.8", PublishedAt: 2019-08-29T00:00:00, Value: 0 }, { Name: "barryvdh/laravel-ide-helper", Version: "v2.6.6", PublishedAt: 2019-10-30T00:00:00, Value: 0 }, { Name: "barryvdh/reflection-docblock", Version: "v2.0.6", PublishedAt: 2018-12-13T00:00:00, Value: 0 }, { Name: "composer/ca-bundle", Version: "1.2.5", PublishedAt: 2019-12-11T00:00:00, Value: 0 }, { Name: "composer/composer", Version: "1.9.1", PublishedAt: 2019-11-01T00:00:00, Value: 0 }, { Name: "composer/semver", Version: "1.5.0", PublishedAt: 2019-03-19T00:00:00, Value: 0 }, { Name: "composer/spdx-licenses", Version: "1.5.2", PublishedAt: 2019-07-29T00:00:00, Value: 0 }, { Name: "composer/xdebug-handler", Version: "1.4.0", PublishedAt: 2019-11-06T00:00:00, Value: 0 }, { Name: "doctrine/instantiator", Version: "1.3.0", PublishedAt: 2019-10-21T00:00:00, Value: 0 }, { Name: "facade/flare-client-php", Version: "1.3.1", PublishedAt: 2019-12-15T00:00:00, Value: 0 }, { Name: "facade/ignition", Version: "1.13.0", PublishedAt: 2019-11-27T00:00:00, Value: 0 }, { Name: "facade/ignition-contracts", Version: "1.0.0", PublishedAt: 2019-08-30T00:00:00, Value: 0 }, { Name: "filp/whoops", Version: "2.7.0", PublishedAt: 2019-12-29T00:00:00, Value: 0.010958904109589041 }, { Name: "fzaninotto/faker", Version: "v1.9.1", PublishedAt: 2019-12-12T00:00:00, Value: 0 }, { Name: "hamcrest/hamcrest-php", Version: "v2.0.0", PublishedAt: 2016-01-20T00:00:00, Value: 0 }, { Name: "jakub-onderka/php-console-color", Version: "v0.2", PublishedAt: 2018-09-29T00:00:00, Value: 0 }, { Name: "jakub-onderka/php-console-highlighter", Version: "v0.4", PublishedAt: 2018-09-29T00:00:00, Value: 0 }, { Name: "justinrainbow/json-schema", Version: "5.2.9", PublishedAt: 2019-09-25T00:00:00, Value: 0 }, { Name: "laravel/browser-kit-testing", Version: "v5.1.3", PublishedAt: 2019-07-30T00:00:00, Value: 0 }, { Name: "maximebf/debugbar", Version: "v1.16.1", PublishedAt: 2019-11-24T00:00:00, Value: 0.16712328767123288 }, { Name: "mockery/mockery", Version: "1.3.1", PublishedAt: 2019-12-26T00:00:00, Value: 0 }, { Name: "myclabs/deep-copy", Version: "1.9.4", PublishedAt: 2019-12-15T00:00:00, Value: 0 }, { Name: "nunomaduro/collision", Version: "v3.0.1", PublishedAt: 2019-03-07T00:00:00, Value: 0 }, { Name: "phar-io/manifest", Version: "1.0.3", PublishedAt: 2018-07-08T00:00:00, Value: 0 }, { Name: "phar-io/version", Version: "2.0.1", PublishedAt: 2018-07-08T00:00:00, Value: 0 }, { Name: "phpdocumentor/reflection-common", Version: "2.0.0", PublishedAt: 2018-08-07T00:00:00, Value: 0 }, { Name: "phpdocumentor/reflection-docblock", Version: "4.3.4", PublishedAt: 2019-12-28T00:00:00, Value: 0.021917808219178082 }, { Name: "phpdocumentor/type-resolver", Version: "1.0.1", PublishedAt: 2019-08-22T00:00:00, Value: 0 }, { Name: "phploc/phploc", Version: "5.0.0", PublishedAt: 2019-03-16T00:00:00, Value: 0 }, { Name: "phpspec/prophecy", Version: "1.10.1", PublishedAt: 2019-12-22T00:00:00, Value: 0 }, { Name: "phpunit/php-code-coverage", Version: "7.0.10", PublishedAt: 2019-11-20T00:00:00, Value: 0 }, { Name: "phpunit/php-file-iterator", Version: "2.0.2", PublishedAt: 2018-09-13T00:00:00, Value: 0 }, { Name: "phpunit/php-text-template", Version: "1.2.1", PublishedAt: 2015-06-21T00:00:00, Value: 0 }, { Name: "phpunit/php-timer", Version: "2.1.2", PublishedAt: 2019-06-07T00:00:00, Value: 0 }, { Name: "phpunit/php-token-stream", Version: "3.1.1", PublishedAt: 2019-09-17T00:00:00, Value: 0 }, { Name: "phpunit/phpunit", Version: "8.5.1", PublishedAt: 2019-12-25T00:00:00, Value: 0 }, { Name: "scrivo/highlight.php", Version: "v9.12.0", PublishedAt: 2017-11-15T00:00:00, Value: -2.0767123287671234 }, { Name: "sebastian/code-unit-reverse-lookup", Version: "1.0.1", PublishedAt: 2017-03-04T00:00:00, Value: 0 }, { Name: "sebastian/comparator", Version: "3.0.2", PublishedAt: 2018-07-12T00:00:00, Value: 0 }, { Name: "sebastian/diff", Version: "3.0.2", PublishedAt: 2019-02-04T00:00:00, Value: 0 }, { Name: "sebastian/environment", Version: "4.2.3", PublishedAt: 2019-11-20T00:00:00, Value: 0 }, { Name: "sebastian/exporter", Version: "3.1.2", PublishedAt: 2019-09-14T00:00:00, Value: 0 }, { Name: "sebastian/finder-facade", Version: "1.2.2", PublishedAt: 2017-11-18T00:00:00, Value: 0 }, { Name: "sebastian/global-state", Version: "3.0.0", PublishedAt: 2019-02-01T00:00:00, Value: 0 }, { Name: "sebastian/object-enumerator", Version: "3.0.3", PublishedAt: 2017-08-03T00:00:00, Value: 0 }, { Name: "sebastian/object-reflector", Version: "1.1.1", PublishedAt: 2017-03-29T00:00:00, Value: 0 }, { Name: "sebastian/recursion-context", Version: "3.0.0", PublishedAt: 2017-03-03T00:00:00, Value: 0 }, { Name: "sebastian/resource-operations", Version: "2.0.1", PublishedAt: 2018-10-04T00:00:00, Value: 0 }, { Name: "sebastian/type", Version: "1.1.3", PublishedAt: 2019-07-02T00:00:00, Value: 0 }, { Name: "sebastian/version", Version: "2.0.1", PublishedAt: 2016-10-03T00:00:00, Value: 0 }, { Name: "seld/jsonlint", Version: "1.7.2", PublishedAt: 2019-10-24T00:00:00, Value: 0 }, { Name: "seld/phar-utils", Version: "1.0.1", PublishedAt: 2015-10-13T00:00:00, Value: 0 }, { Name: "squizlabs/php_codesniffer", Version: "3.5.3", PublishedAt: 2019-12-04T00:00:00, Value: 0 }, { Name: "symfony/dom-crawler", Version: "v5.0.2", PublishedAt: 2019-11-18T00:00:00, Value: 0.0547945205479452 }, { Name: "symfony/filesystem", Version: "v5.0.2", PublishedAt: 2019-11-26T00:00:00, Value: 0 }, { Name: "theseer/fdomdocument", Version: "1.6.6", PublishedAt: 2017-06-30T00:00:00, Value: 0 }, { Name: "theseer/tokenizer", Version: "1.1.3", PublishedAt: 2019-06-13T00:00:00, Value: 0 }, { Name: "webmozart/assert", Version: "1.6.0", PublishedAt: 2019-11-24T00:00:00, Value: 0 }, { Name: "wnx/laravel-stats", Version: "v2.0.0", PublishedAt: 2019-09-01T00:00:00, Value: 0 },  ], Total: 4.717808219178083 } }
